radar-prmt-ios/auth/QRCodeViewController.swift

```diff
@@ -12,58 +12,91 @@ import AVFoundation
 import RxSwift
 import os.log
 
-class QRCodeViewController: UIViewController, AVCaptureMetadataOutputObjectsDelegate {
-    @IBOutlet weak var qrPreview: QRPreviewView!
-
-    let captureSession: AVCaptureSession
-    let processingQueue: DispatchQueue
+class QRCodeViewController: UIViewController {
     let disposeBag = DisposeBag()
 
-    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
-        captureSession = AVCaptureSession()
-        processingQueue = DispatchQueue(label: "QR-processing", qos: .userInitiated)
-        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+//    @IBOutlet weak var statusLabel: UILabel!
+    @IBOutlet weak var qrPreview: QRPreviewView! {
+        didSet {
+            qrPreview.delegate = self
+        }
     }
-
-    required init?(coder aDecoder: NSCoder) {
-        captureSession = AVCaptureSession()
-        processingQueue = DispatchQueue(label: "QR-processing", qos: .userInitiated)
-        super.init(coder: aDecoder)
+    
+    var qrData: QRData? = nil {
+        didSet {
+            if qrData != nil {
+                guard let url = URL(string: (qrData?.codeString)!) else {
+                    return
+                }
+                appDelegate.authController.login(to: url)
+                    .subscribeOn(MainScheduler.instance)
+                    .subscribe(weak: self, onNext: { weakSelf in { mpauth in
+                        os_log("Retrieved MetaToken")
+                        DispatchQueue.main.async {
+                            weakSelf.performSegue(withIdentifier: "mainFromQr", sender: self)
+                        }
+                    }}, onError: { weakSelf in { error in
+                        os_log("Failed to retrieve MetaToken: %@", error.localizedDescription)
+                        DispatchQueue.main.async {
```

Comment:
Using the `.subscribeOn(MainScheduler.instance)` should already have this running in the `main` queue.